components/datetime/src/pattern/names.rs

```diff
@@ -2790,6 +2794,180 @@ impl<FSet: DateTimeNamesMarker> RawDateTimeNames<FSet> {
+            needs_decimal_formatter,
+        })
+    }
+
+    #[must_use]
+    pub(crate) fn load_time_zone_field_v_names(
+        &mut self,
+        zone_essentials_provider: &(impl BoundDataProvider<tz::EssentialsV1> + ?Sized),
+        locations_provider: &(impl BoundDataProvider<tz::LocationsV1> + ?Sized),
+        locations_root_provider: &(impl BoundDataProvider<tz::LocationsRootV1> + ?Sized),
+        mz_generic_short_provider: &(impl BoundDataProvider<tz::MzGenericShortV1> + ?Sized),
+        mz_period_provider: &(impl BoundDataProvider<tz::MzPeriodV1> + ?Sized),
+        prefs: DateTimeFormatterPreferences,
+    ) -> Result<LoadTimeZoneNamesByFieldResult, PatternLoadError> {
+        let needs_decimal_formatter = true;
+        self.load_time_zone_essentials(zone_essentials_provider, prefs)?;
+        self.load_time_zone_generic_short_names(
+            mz_generic_short_provider,
+            mz_period_provider,
+            prefs,
+        )?;
+        // For fallback:
+        self.load_time_zone_location_names(locations_provider, locations_root_provider, prefs)?;
+        Ok(LoadTimeZoneNamesByFieldResult {
+            needs_decimal_formatter,
+        })
+    }
+
+    #[must_use]
+    pub(crate) fn load_time_zone_field_vvvv_names(
+        &mut self,
+        zone_essentials_provider: &(impl BoundDataProvider<tz::EssentialsV1> + ?Sized),
+        locations_provider: &(impl BoundDataProvider<tz::LocationsV1> + ?Sized),
+        locations_root_provider: &(impl BoundDataProvider<tz::LocationsRootV1> + ?Sized),
+        mz_generic_long_provider: &(impl BoundDataProvider<tz::MzGenericLongV1> + ?Sized),
+        mz_standard_long_provider: &(impl BoundDataProvider<tz::MzStandardLongV1> + ?Sized),
+        mz_period_provider: &(impl BoundDataProvider<tz::MzPeriodV1> + ?Sized),
+        prefs: DateTimeFormatterPreferences,
+    ) -> Result<LoadTimeZoneNamesByFieldResult, PatternLoadError> {
+        let needs_decimal_formatter = true;
+        self.load_time_zone_essentials(zone_essentials_provider, prefs)?;
+        self.load_time_zone_generic_long_names(
+            mz_generic_long_provider,
+            mz_standard_long_provider,
+            mz_period_provider,
+            prefs,
+        )?;
+        // For fallback:
+        self.load_time_zone_location_names(locations_provider, locations_root_provider, prefs)?;
+        Ok(LoadTimeZoneNamesByFieldResult {
+            needs_decimal_formatter,
+        })
+    }
+
+    #[must_use]
+    #[allow(non_snake_case)] // this is a private function named after the case-sensitive CLDR field
+    pub(crate) fn load_time_zone_field_V_names(
+        &mut self,
+        _prefs: DateTimeFormatterPreferences,
+    ) -> Result<LoadTimeZoneNamesByFieldResult, PatternLoadError> {
+        // no data required
+        let needs_decimal_formatter = false;
+        Ok(LoadTimeZoneNamesByFieldResult {
+            needs_decimal_formatter,
+        })
+    }
+
+    #[must_use]
```

Comment:
nit: results are already must_use, this is not necessary

Reply:
yeah clippy caught that. It was intended to make people handle the decimal loading, but it didn't actually do that.